blog post about kiali deprecation (#888)
kyma-project/website: content/blog-posts/2022-09-30-Kiali-deprecation/index.md

---
title: "Kiali deprecation"
author:
  name: Andreas Thaler, PO @Kyma"
tags:
  - kyma
  - observability
  - Kiali

redirectFrom:
  - "/blog/kiali-deprecation"
---

> I’m part of the Kyma team working on the Observability feature, and I like to inform you today that we decided to deprecate the Kyma Kiali component and like t to provide you some reasoning behind that decision.

Reviewer: ```suggestion
As part of the Kyma team working on the Observability feature, I'd like to let you know that we decided to deprecate the Kyma Kiali component. In this blog post, I'd like to provide you our reasoning behind that decision.
```

> As part of the Kyma team working on the Observability feature, I'd like to let you know that we decided to deprecate the Kyma Kiali component. In this blog post, I'm going to give you our reasoning behind that decision.

Reviewer 2: I would change "Observability feature" to "Observability capabilites" since it's not really a single feature.

> As part of the Kyma team working on the Observability feature, I'd like to let you know that we decided to deprecate the Kyma Kiali component. In this blog post, I'd like to provide you our reasoning behind that decision.

Reviewer: ```suggestion
As part of the Kyma team working on the Observability feature, I'd like to let you know that we decided to deprecate the Kyma Kiali component. In this blog post, I'm going to give you our reasoning behind that decision.
```
didn't notice the Wortwiederholung before

> ## Motivation

Reviewer: ```suggestion
## The Background
```

> In the recent [blog post](https://kyma-project.io/blog/2022/9/21/observability-strategy/) I outlined the new focus for the Observability domain in Kyma. To summarize, the shift into providing enterprise-grade qualities for Kyma modules moved the focus to enable the user streaming telemetry data into existing observability backends. Hereby, Kyma explicitly will not focus on being yet another way of running a specific observability stacks in enterprise-grade qualities but focusing on an open and easy integration into existing solutions.

Reviewer: ```suggestion
In my recent [blog post](https://kyma-project.io/blog/2022/9/21/observability-strategy/), I outlined the new strategy for the Observability domain in Kyma. To summarize, the shift towards providing enterprise-grade qualities for Kyma modules moved the focus to enable the users to stream telemetry data into their existing observability backends. Hereby, Kyma explicitly decided to support open and easy integration into existing solutions instead of being yet another provider of a specific observability stack in enterprise-grade qualities.
```
Just guessing :D 
- I tried to get rid of the repeated use of "focus" but during rephrasing, I may have misrepresented which parts you want to stress.
- I wanted to use the word "strategy" so readers easily recognize which blog post you mean, even without clicking the link

> In my recent [blog post](https://kyma-project.io/blog/2022/9/21/observability-strategy/), I outlined the new strategy for the Observability domain in Kyma. To summarize, the shift towards providing enterprise-grade qualities for Kyma modules moved the focus to enable the users to stream telemetry data into their existing observability backends. Hereby, Kyma explicitly decided to support open and easy integration into existing solutions instead of being yet another provider of a specific observability stack in enterprise-grade quality.

Reviewer 2: I would rephrase the summary more like:
"To summarize: the shift towards providing enterprise-grade qualities for Kyma modules changed the focus to enabling users to stream telemetry data into centralized enterprise-grade observability backends.

> In my recent [blog post](https://kyma-project.io/blog/2022/9/21/observability-strategy/), I outlined the new strategy for the Observability domain in Kyma. To summarize, the shift towards providing enterprise-grade qualities for Kyma modules moved the focus to enable the users to stream telemetry data into their existing observability backends. Hereby, Kyma explicitly decided to support open and easy integration into existing solutions instead of being yet another provider of a specific observability stack in enterprise-grade qualities.

Reviewer: ```suggestion
In my recent [blog post](https://kyma-project.io/blog/2022/9/21/observability-strategy/), I outlined the new strategy for the Observability domain in Kyma. To summarize, the shift towards providing enterprise-grade qualities for Kyma modules moved the focus to enable the users to stream telemetry data into their existing observability backends. Hereby, Kyma explicitly decided to support open and easy integration into existing solutions instead of being yet another provider of a specific observability stack in enterprise-grade quality.
```

> Now, we'll apply the same principle to Kiali. Kyma's current Kiali component is very lightweight and does not provide well-defined configuration options at runtime. The Kyma team supports the Kiali project, but doesn't want to become an active contributor with the goal of running Kiali with managed qualities inside a Kyma cluster. Instead, Kyma wants to assure that the users can connect with Kiali whenever they want.

Reviewer 2: I would rephrase this a bit to put the focus on how running an in-cluster component doesn't fit the strategy:
"Taking into consideration that, currently, the Kiali component is not enterprise-grade quality and is not centralized, we concluded that the current approach does not fit our strategy. Thus, we want to enable users to easily connect all their Kyma clusters to a central Kiali (or other) instance rather than maintain individual instances in each cluster."

Author: The argumentation with recommending a managed offering does not really apply to kiali. I cannot find something like a managed offering (as it requires the metrics in a prometheus and access to the apiserver). So usually you would deploy it along with istio inside the cluster. So I would keep the same argumentation that it is simply not our focus to provide such service in high quality.

> The same principle we will apply now to Kiali. The current Kiali component is very lightweight and does not provide well-defined configuration options at runtime. The Kyma team supports the Kiali project, but don't want to become an active contributor with the goal of running Kiali with managed qualities inside a Kyma cluster. Instead, Kyma needs to assure that the user is at anytime enabled to integrate with Kiali, if they want to.

Reviewer: ```suggestion
Now, we'll apply the same principle to Kiali. Kyma's current Kiali component is very lightweight and does not provide well-defined configuration options at runtime. The Kyma team supports the Kiali project, but doesn't want to become an active contributor with the goal of running Kiali with managed qualities inside a Kyma cluster. Instead, Kyma wants to assure that the users can connect with Kiali whenever they want.
```

> ## The Facts & Alternatives

Reviewer: ```suggestion
## The Facts and Alternatives
```

> ## The Facts & Alternatives

Reviewer: ```suggestion
## The Future
```

> The Kiali component will be marked as deprecated with Kyma version 2.8. Instructions on how to install Kiali on your own are already [provided](https://github.com/kyma-project/examples/tree/main/kiali) and maintained on a best effort base. After you transitioned to a custom installation, the Kyma component gets removed with Kyma 2.11.

Reviewer: ```suggestion
The Kiali component will be marked as deprecated with Kyma version 2.8. 
Instructions on [how to install Kiali on your own](https://github.com/kyma-project/examples/tree/main/kiali) have already been provided and will be maintained on a best-effort base. After you transitioned to a custom Kiali installation, Kyma's Kiali component is planned to be removed with Kyma 2.11.
```

> ### Conclusion

Reviewer: ```suggestion
```

> To sharpen the focus of the project the Kiali component gets transformed into a simple set of instructions. You can start the transformation already today.

Reviewer: ```suggestion
To sharpen the focus of the Kyma project, the Kiali component is transformed into a simple set of instructions. You can start the transformation already today.
```
